### PR TITLE
F#src 3068 fix ssh bug

### PR DESCRIPTION
**Pull request by ToivoHn**
All molecule scenarios pass (apart from default_ec2), and this has been tested with real hardware as well (control-computer, teleop-server) to verify ssh keys really does work (without password)

### CODE REVIEW

File: ansible/playbooks/molecule/teleop_empty_server_docker/tests/test_teleop_server.py [hunk 1]
```diff
@@ -33,18 +33,24 @@ def test_docker_container_exists(host):
...
         '/home/user/projects/shadow_robot/base/src/sr_config')
     assert stat['size'] > 0
 
 
 def test_icons_in_docker(host):
-    f = host.file('/home/Desktop/Teleop_Launcher.desktop')
-    assert f.exists
+    user = host.user().name
+    icon_list = []
+    desktop_path = '/home/'+str(user)+'/Desktop/'
```
Comment (AndriyPt) [inline ansible/playbooks/molecule/teleop_empty_server_docker/tests/test_teleop_server.py]: please add spaces before and after plus sign

File: ansible/roles/products/teleop/control-machine/deploy/tasks/main.yml [hunk 2]
```diff
@@ -47,3 +47,25 @@
       iface {{ ur_interface }} inet static
         address 192.168.1.100
         netmask 255.255.255.0
+
+- name: Check if the ssh public key has been generated in ~/.ssh/ on machine running Ansible (teleop server)
+  stat:
+    path: "~/.ssh/id_rsa.pub"
```
Comment (AndriyPt) [inline ansible/roles/products/teleop/control-machine/deploy/tasks/main.yml]: please use {{user_folder}}

File: ansible/roles/installation/docker/tasks/main.yml [hunk 3]
```diff
@@ -12,6 +12,22 @@
   when: pip_status.rc != 0
   become: yes
 
+- name: Check the default python3 version
```
Comment (AndriyPt) [inline ansible/roles/installation/docker/tasks/main.yml]: please remove

File: ansible/roles/products/teleop/server/deploy/tasks/main.yml [hunk 4]
```diff
@@ -1,25 +1,13 @@
...
-    dest: "/home/{{ user }}/.ssh/{{ item }}"
-    mode: 0600
-    state: present
-  with_items:
-    - id_rsa
-    - id_rsa.pub
+- name: Evaluating the SSH agent and adding the ssh identity
+  shell: |
+    eval "$(ssh-agent)"
+    ssh-add ~/.ssh/id_rsa
```
Comment (AndriyPt) [inline ansible/roles/products/teleop/server/deploy/tasks/main.yml]: please use {{ user_home }}

File: ansible/roles/products/teleop/server/desktop-icons/defaults/main.yml [hunk 5]
```diff
@@ -5,3 +5,5 @@ shadow_hand_launcher_folder: "{{ user_folder }}/.shadow_launcher_app/shadow_hand
 save_latest_ros_logs_folder: "{{ user_folder }}/.shadow_save_log_app/save_latest_ros_logs"
 container_name : "teleop-server"
 sr_config_branch: "demohand_A"
+nuc_address: "10.10.10.10"
```
Comment (AndriyPt) [inline ansible/roles/products/teleop/server/desktop-icons/defaults/main.yml]: please remove

File: ansible/roles/products/teleop/server/deploy/defaults/main.yml [hunk 6]
```diff
@@ -1,9 +1,13 @@
...
 reinstall: false
-container_name: "teleop-server"
+container_name: "teleop"
 nvidia_docker: false
 cyberglove: "A-penny"
 use_aws: false
 use_openvpn: false
+user: "{{ ansible_user_id }}"
+user_folder: "/home/{{ user }}"
+nuc_address: "10.10.10.10"
```
Comment (AndriyPt) [inline ansible/roles/products/teleop/server/deploy/defaults/main.yml]: please remove

File: bin/run-ansible.sh [hunk 7]
```diff
@@ -104,6 +104,10 @@ echo ""
 
 pushd $aurora_home
 
+# Make sure pip3 default version matches the python3 version (3.5, 3.6, or 3.7 etc.)
```
Comment (AndriyPt) [inline bin/run-ansible.sh]: please remove

File: ansible/playbooks/molecule/teleop_empty_server_docker/tests/test_teleop_server.py (hunk 1 above)
Comment (ToivoHn) [reply ansible/playbooks/molecule/teleop_empty_server_docker/tests/test_teleop_server.py]: Fixed in the last commit

File: ansible/roles/products/teleop/control-machine/deploy/tasks/main.yml (hunk 2 above)
Comment (ToivoHn) [reply ansible/roles/products/teleop/control-machine/deploy/tasks/main.yml]: Fixed in the last commit

File: ansible/roles/installation/docker/tasks/main.yml (hunk 3 above)
Comment (ToivoHn) [reply ansible/roles/installation/docker/tasks/main.yml]: Fixed in the last commit

File: ansible/roles/products/teleop/server/deploy/tasks/main.yml (hunk 4 above)
Comment (ToivoHn) [reply ansible/roles/products/teleop/server/deploy/tasks/main.yml]: Fixed in the last commit

File: ansible/roles/products/teleop/server/desktop-icons/defaults/main.yml (hunk 5 above)
Comment (ToivoHn) [reply ansible/roles/products/teleop/server/desktop-icons/defaults/main.yml]: Fixed in the last commit

File: ansible/roles/products/teleop/server/deploy/defaults/main.yml (hunk 6 above)
Comment (ToivoHn) [reply ansible/roles/products/teleop/server/deploy/defaults/main.yml]: Fixed in the last commit

File: bin/run-ansible.sh (hunk 7 above)
Comment (ToivoHn) [reply bin/run-ansible.sh]: Fixed in the last commit

**Pull request merged**